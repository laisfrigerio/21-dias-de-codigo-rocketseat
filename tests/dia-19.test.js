const { JSDOM } = require('jsdom')
const fs = require('fs')
const path = require('path')

const html = fs.readFileSync(path.resolve(__dirname, '../dia-19/index.html'), 'utf8')

let dom
let container

describe('dia 19', () => {
  beforeEach(() => {
    dom = new JSDOM(html, { runScripts: 'dangerously' })
    container = dom.window.document.body
  })

  test('should match the random image API challenge', () => {
    expect(container).toMatchSnapshot();
  });
})