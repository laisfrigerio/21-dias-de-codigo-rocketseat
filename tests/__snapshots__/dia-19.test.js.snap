// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`dia 19 should match the random image API challenge 1`] = `
<body
  class="main-container"
  id="particles-js"
>
  
  
  <header />
  
  
  <div
    class="glassmorphism-card"
  >
    
    
    <div
      class="card"
    >
      
      
      <a
        href="https://laisfrigerio.com.br/"
        title="Random Dog Picture"
      >
        
        
        <img
          alt="Avatar"
          class="avatar"
          src="data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAAGQAAABkCAYAAABw4pVUAAAAAXNSR0IArs4c6QAABllJREFUeF7tnFtMVFcYhReicvFBHhBighYoiBGHQHmRodYLlQBNrBghUEmEQlAId6RgR+xgoVgGtSgWa4ytjqXKAOWioEBFTEix1AeppfJgTDChEaka64MgYrN344TLTOdksOFv+p+EB2bWPmft9Z21zz48YOOnUr0EH2QSsGEgZFhIIwyEFg8GQowHA2Eg1BIg5oefIQyEWALE7HBDGAixBIjZ4YYwEGIJELPDDWEgxBIgZocbwkCIJUDMDjeEgRBLgJgdbggDIZYAMTvcEAZCLAFidrghDIRYAsTscEMYCLEEiNnhhjAQYgkQs8MNYSDEEiBmhxvCQIglQMwON4SBEEuAmB1uCAMhlgAxO9wQBkIsAWJ2uCEMhFgCxOxwQxgIsQSI2XktDVm8eDGKi4vxzrp1iI6OxsDt21Om6eDggKzsbISGhmLRokX4rb8fZTodfr11y6hTonld2Xl7e6P84EE4OzsjWK2ecdqgoCCkZ2TAy8sLfz55gsbGRhw7dgwvXrwwapVorPE7ayAqlQq68nI8ffoUYqKmgOh0OvgHBEBXVobhBw/wQWws1MHB2BoZieHhYelbicaaCU4fsyUyEgUFBRgaGoKrq+sMID4+Pvi2uhpNTU1obGiA27JlUl9fX4/Dhw7J0ynRWOt11kAqjhzBL3196O3txRm9fgaQ5cuXo/nCBWSkp6Orq0v6nD9/Pi62tKDl4kVUVFRAicbaCU4fd7mtDZ/u348VPj5ITEycAeSz0lJ4eHggNibGODQqOhp5eXnYuGGDvPGUaKz1O2sg4i67f/8+/Pz8oD97dgaQbVFR8g4LWrMGz58/N/rUarXwWblSTlyJZvoEw8PDZTAxMTHGJdLf3x/fnD6N3bm56OjoMJnJK78fJiaaBPLDlSuoNRhQVVVlHC/GtLW3Iz09Hde6uqBEM2dAXl3YHJDc3FxsDAnBexERUzwmJSVBhKIOCoISjakJinY6OTkhfscO2NjY4Ltz5zA4OIi83bst5mEKiKOjI37s6YFGo8GF5uYp5/j5xg3Z5rraWosa/ZkzFq9vTjDrhlgC8olWC99Vq2RzJh+xsbEo2LMHAf7+KNy3z6JmYmJixhyWLFmC7xsa5PPHwd4eKampiNyyBQ8fPrQYiCkgLi4uaO/oQFZmJjo7O6eco+vaNdScPw+DwWBRIzYA1h7/aSBi0pFbtyIzMxO2trYoKS7GpUuXFGXxvwWSnZODTZs2ISI8fEpQycnJiI+Ph1qthhKNuZTt7OzQefUqxsfH8W5ICMbGxqwGIrbePdevY69Gg+ZJS5ZYDsWS9cXhw6itrbWo0ev1ijyYEv3rDRF3cGFhoXyoj46OGj2UlJTgDXd3xG3fLu9ySxpzM8zJyZHvPwsWLEDb5ctynVdymHuoi4e32PJWHj1qPI2bm5vcFaampKC7u1s+4C1plHiYEyBLly5FS2srPsrLQ3t7u/Rgb28Psf2srq7GV8ePQ4nGlHlfX1+5sxNts1u4EEcrKyXg/v5+i3mYAyJ2f6tVKkRt24aXL//+74eiyTt37cKG9evx7NkzKNFYNGBGMKuGzJs3D28FBspTv+npiY81GhRptRi8dw9jo6Po6+uT3xUVFeHttWvx+YEDGBkZQUJCAnxXr5Yvho8fP1asmTwH8S4jdlUDAwNyiRGHeEF1d3eXW2mxhE0/xF8UvFeskB9HRETIn7S0NPn7HyMjuHv3rhxfYzCgtbUV9XV18PD0RH5+Pr4+dQonTpyQWiWaOQEi1u+fentNXvv3oSGEhYXJ74QuMytLBiC2ln03b6K0tBR37twxjlWimXyh5J07ERcXh/c3b8ajR4/kV2LXJf7MIV5Qj096j3g1Ljg4GF+a+Fx8L5agwr17pTQwMFBuxQU8ccMYampw8uRJTN7pKdFYA2VWDbHmgjzmnxNgIMTuEAbCQIglQMwON4SBEEuAmB1uCAMhlgAxO9wQBkIsAWJ2uCEMhFgCxOxwQxgIsQSI2eGGMBBiCRCzww1hIMQSIGaHG8JAiCVAzA43hIEQS4CYHW4IAyGWADE73BAGQiwBYna4IQyEWALE7HBDGAixBIjZ4YYwEGIJELPDDWEgxBIgZocbwkCIJUDMDjeEgRBLgJgdbggDIZYAMTvcEAZCLAFidrghDIRYAsTscEMYCLEEiNn5Cx0QdU6P8ZO+AAAAAElFTkSuQmCC"
        />
        
      
      </a>
      
      
      <h1>
        Random Dog Picture
      </h1>
      
      
      <ul
        class="social-network"
      >
        
        
        <li
          class="ico"
          title="Meu Instagram"
        >
          
          
          <a
            href="https://www.instagram.com/laisfrigerio"
          >
            
            
            <svg
              role="img"
              viewBox="0 0 24 24"
              xmlns="http://www.w3.org/2000/svg"
            >
              
              
              <title>
                Instagram icon
              </title>
              
              
              <path
                d="M12 0C8.74 0 8.333.015 7.053.072 5.775.132 4.905.333 4.14.63c-.789.306-1.459.717-2.126 1.384S.935 3.35.63 4.14C.333 4.905.131 5.775.072 7.053.012 8.333 0 8.74 0 12s.015 3.667.072 4.947c.06 1.277.261 2.148.558 2.913.306.788.717 1.459 1.384 2.126.667.666 1.336 1.079 2.126 1.384.766.296 1.636.499 2.913.558C8.333 23.988 8.74 24 12 24s3.667-.015 4.947-.072c1.277-.06 2.148-.262 2.913-.558.788-.306 1.459-.718 2.126-1.384.666-.667 1.079-1.335 1.384-2.126.296-.765.499-1.636.558-2.913.06-1.28.072-1.687.072-4.947s-.015-3.667-.072-4.947c-.06-1.277-.262-2.149-.558-2.913-.306-.789-.718-1.459-1.384-2.126C21.319 1.347 20.651.935 19.86.63c-.765-.297-1.636-.499-2.913-.558C15.667.012 15.26 0 12 0zm0 2.16c3.203 0 3.585.016 4.85.071 1.17.055 1.805.249 2.227.415.562.217.96.477 1.382.896.419.42.679.819.896 1.381.164.422.36 1.057.413 2.227.057 1.266.07 1.646.07 4.85s-.015 3.585-.074 4.85c-.061 1.17-.256 1.805-.421 2.227-.224.562-.479.96-.899 1.382-.419.419-.824.679-1.38.896-.42.164-1.065.36-2.235.413-1.274.057-1.649.07-4.859.07-3.211 0-3.586-.015-4.859-.074-1.171-.061-1.816-.256-2.236-.421-.569-.224-.96-.479-1.379-.899-.421-.419-.69-.824-.9-1.38-.165-.42-.359-1.065-.42-2.235-.045-1.26-.061-1.649-.061-4.844 0-3.196.016-3.586.061-4.861.061-1.17.255-1.814.42-2.234.21-.57.479-.96.9-1.381.419-.419.81-.689 1.379-.898.42-.166 1.051-.361 2.221-.421 1.275-.045 1.65-.06 4.859-.06l.045.03zm0 3.678c-3.405 0-6.162 2.76-6.162 6.162 0 3.405 2.76 6.162 6.162 6.162 3.405 0 6.162-2.76 6.162-6.162 0-3.405-2.76-6.162-6.162-6.162zM12 16c-2.21 0-4-1.79-4-4s1.79-4 4-4 4 1.79 4 4-1.79 4-4 4zm7.846-10.405c0 .795-.646 1.44-1.44 1.44-.795 0-1.44-.646-1.44-1.44 0-.794.646-1.439 1.44-1.439.793-.001 1.44.645 1.44 1.439z"
                fill="#ffffff"
              />
              
            
            </svg>
            
          
          </a>
          
        
        </li>
        
        
        <li
          class="ico"
          title="Meu Github"
        >
          
          
          <a
            href="https://github.com/laisfrigerio"
          >
            
            
            <svg
              role="img"
              viewBox="0 0 24 24"
              xmlns="http://www.w3.org/2000/svg"
            >
              
              
              <title>
                GitHub icon
              </title>
              
              
              <path
                d="M12 .297c-6.63 0-12 5.373-12 12 0 5.303 3.438 9.8 8.205 11.385.6.113.82-.258.82-.577 0-.285-.01-1.04-.015-2.04-3.338.724-4.042-1.61-4.042-1.61C4.422 18.07 3.633 17.7 3.633 17.7c-1.087-.744.084-.729.084-.729 1.205.084 1.838 1.236 1.838 1.236 1.07 1.835 2.809 1.305 3.495.998.108-.776.417-1.305.76-1.605-2.665-.3-5.466-1.332-5.466-5.93 0-1.31.465-2.38 1.235-3.22-.135-.303-.54-1.523.105-3.176 0 0 1.005-.322 3.3 1.23.96-.267 1.98-.399 3-.405 1.02.006 2.04.138 3 .405 2.28-1.552 3.285-1.23 3.285-1.23.645 1.653.24 2.873.12 3.176.765.84 1.23 1.91 1.23 3.22 0 4.61-2.805 5.625-5.475 5.92.42.36.81 1.096.81 2.22 0 1.606-.015 2.896-.015 3.286 0 .315.21.69.825.57C20.565 22.092 24 17.592 24 12.297c0-6.627-5.373-12-12-12"
                fill="#ffffff"
              />
              
            
            </svg>
            
          
          </a>
          
        
        </li>
        
        
        <li
          class="ico"
          title="Meu Canal no Youtube"
        >
          
          
          <a
            href="https://www.youtube.com/user/12345678912345678432"
          >
            
            
            <!--?xml version="1.0" encoding="iso-8859-1"?-->
            
            
            
            <svg
              height="97.75px"
              id="Capa_1"
              style="enable-background:new 0 0 97.75 97.75;"
              version="1.1"
              viewBox="0 0 97.75 97.75"
              width="97.75px"
              x="0px"
              xml:space="preserve"
              xmlns="http://www.w3.org/2000/svg"
              xmlns:xlink="http://www.w3.org/1999/xlink"
              y="0px"
            >
              
              
              <g>
                
                
                <g>
                  
                  
                  <path
                    d="M39.969,59.587c7.334-3.803,14.604-7.571,21.941-11.376c-7.359-3.84-14.627-7.63-21.941-11.447
			C39.969,44.398,39.969,51.954,39.969,59.587z"
                    fill="#ffffff"
                  />
                  
                  
                  <path
                    d="M48.875,0C21.883,0,0,21.882,0,48.875S21.883,97.75,48.875,97.75S97.75,75.868,97.75,48.875S75.867,0,48.875,0z
			 M82.176,65.189c-0.846,3.67-3.848,6.377-7.461,6.78c-8.557,0.957-17.217,0.962-25.842,0.957c-8.625,0.005-17.287,0-25.846-0.957
			c-3.613-0.403-6.613-3.11-7.457-6.78c-1.203-5.228-1.203-10.933-1.203-16.314s0.014-11.088,1.217-16.314
			c0.844-3.67,3.844-6.378,7.457-6.782c8.559-0.956,17.221-0.961,25.846-0.956c8.623-0.005,17.285,0,25.841,0.956
			c3.615,0.404,6.617,3.111,7.461,6.782c1.203,5.227,1.193,10.933,1.193,16.314S83.379,59.962,82.176,65.189z"
                    fill="#ffffff"
                  />
                  
                
                </g>
                
              
              </g>
              
              
              <g />
              
              
              <g />
              
              
              <g />
              
              
              <g />
              
              
              <g />
              
              
              <g />
              
              
              <g />
              
              
              <g />
              
              
              <g />
              
              
              <g />
              
              
              <g />
              
              
              <g />
              
              
              <g />
              
              
              <g />
              
              
              <g />
              
            
            </svg>
            

          
          </a>
          
        
        </li>
        
        
        <li
          class="ico"
          title="Meu Linkedin"
        >
          
          
          <a
            href="https://www.linkedin.com/in/laisfrigerio/"
          >
            
            
            <svg
              role="img"
              viewBox="0 0 24 24"
              xmlns="http://www.w3.org/2000/svg"
            >
              
              
              <title>
                LinkedIn icon
              </title>
              
              
              <path
                d="M20.447 20.452h-3.554v-5.569c0-1.328-.027-3.037-1.852-3.037-1.853 0-2.136 1.445-2.136 2.939v5.667H9.351V9h3.414v1.561h.046c.477-.9 1.637-1.85 3.37-1.85 3.601 0 4.267 2.37 4.267 5.455v6.286zM5.337 7.433c-1.144 0-2.063-.926-2.063-2.065 0-1.138.92-2.063 2.063-2.063 1.14 0 2.064.925 2.064 2.063 0 1.139-.925 2.065-2.064 2.065zm1.782 13.019H3.555V9h3.564v11.452zM22.225 0H1.771C.792 0 0 .774 0 1.729v20.542C0 23.227.792 24 1.771 24h20.451C23.2 24 24 23.227 24 22.271V1.729C24 .774 23.2 0 22.222 0h.003z"
                fill="#ffffff"
              />
              
            
            </svg>
            
          
          </a>
          
        
        </li>
        
      
      </ul>
      
    
    </div>
    
  
  </div>
  

  
  <label
    class="theme-mode"
  >
    
    
    <input
      id="slider"
      onchange="toggleTheme()"
      type="checkbox"
    />
    
    
    <span
      class="slider"
    />
    
  
  </label>
  

  
  <footer />
  

  
  <script
    src="https://cdn.jsdelivr.net/npm/particles.js@2.0.0/particles.min.js"
  />
  
  
  <script
    src="../assets/js/theme.js"
  />
  
  
  <script
    src="../assets/js/particles.js"
  />
  
  
  <script
    src="../assets/js/include-html.js"
  />
  
  
  <script
    src="scripts.js"
  />
  



</body>
`;
